### PR TITLE
update user info

### LINKED ISSUE #49: Connect the user profile.
**Is your feature request related to a problem? Please describe.**
N/A

**Describe how the solution should behave**
The user should be able to update their data, user email, and phone number. Once the update is live in firestore, the redux should be updated as well.

**Screenshots of the design/if any**
image

**Additional context**
The user info should be in the User slice once the user is authenticated.

### PR DESCRIPTION
**Pull request by ZhraaBaharAluloom**
## What?
In this PR, created a function that enable user to update his info such as: username and phone number in the firestore and save the changes in redux 

## How to use?
Go to the `profile/user ` then click on edit button to update the username or the phone number

## Screenshots 
Screen Shot 2021-10-25 at 5 42 28 PM
Screen Shot 2021-10-25 at 5 42 36 PM

### CODE REVIEW

File: src/pages/Profile.jsx [hunk 1]
```diff
@@ -18,9 +16,8 @@ function Profile(props) {
   const location = useLocation();
 
   const userName = useSelector(selectorUserName);
-  const userEmail = useSelector(selectorUserEmail);
-  const userPhone = useSelector(selectorUserPhone);
   const userPhoto = useSelector(selectorUserPhoto);
```
Comment (MuhannedNoman) [inline src/pages/Profile.jsx]: The idea that we need multiple Selectors to select data for the same user doesn't make sense!

File: src/store/counter/userSlice.js [hunk 2]
```diff
@@ -110,7 +113,8 @@ const userSlice = createSlice({
   },
 });
 
-export const { setActiveUser, setLogOut } = userSlice.actions;
+export const { setActiveUser, setLogOut, setUpdateUserInfo } =
+  userSlice.actions;
 export const selectorUserName = (state) => state.user.userName;
 export const selectorUserEmail = (state) => state.user.userEmail;
 export const selectorUserPhone = (state) => state.user.userPhone;
```
Comment (MuhannedNoman) [inline src/store/counter/userSlice.js]: Let's just export the user with all of it's data.

File: src/pages/Profile.jsx (hunk 1 above)
Comment (ZhraaBaharAluloom) [reply src/pages/Profile.jsx]: Yes

**Review by MuhannedNoman — APPROVED** [review]
LGTM, nice work here!

**Pull request merged**